### PR TITLE
[FEC-715] Add inline EzButton variant

### PR DESCRIPTION
**Pull request by jmargolisvt**
## What did we change?
Added an `inline` variant to EzButton.

## Why are we doing this?
While transitioning legacy buttons, we discovered a need for a version of the `text` variant that would be used inline.

## Screenshot(s) / Gif(s):
Screenshot 2023-07-14 at 10 01 35 AM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes
 
[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

File: packages/recipe/src/components/EzButton/EzButton.md [hunk 1]
```diff
@@ -32,19 +32,22 @@ Buttons should not:
...
 
+`inline` buttons are `text` buttons that align with surrounding text.
+
 ```jsx
 <EzLayout>
   <EzButton>Filled</EzButton>
   <EzButton variant="outlined">Outlined</EzButton>
   <EzButton variant="text">Text</EzButton>
+  Lorem <EzButton variant="inline">Text</EzButton> ipsum.
 </EzLayout>
```
Comment (noranda) [inline packages/recipe/src/components/EzButton/EzButton.md:51]:
```suggestion
<EzLayout layout="stack">
  <EzLayout>
    <EzButton>Filled</EzButton>
    <EzButton variant="outlined">Outlined</EzButton>
    <EzButton variant="text">Text</EzButton>
  </EzLayout>
  <div>
    This <EzButton variant="inline">Inline</EzButton> button is between text.
  </div>
</EzLayout>
```

File: .changeset/chatty-starfishes-admire.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@ezcater/recipe': patch
+---
+
+Add 'inline' variant for EzButton
```
Comment (noranda) [inline .changeset/chatty-starfishes-admire.md:5]:
```suggestion
feat: add 'inline' variant for EzButton
```

File: .changeset/chatty-starfishes-admire.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@ezcater/recipe': patch
```
Comment (noranda) [inline .changeset/chatty-starfishes-admire.md:2]:
```suggestion
'@ezcater/recipe': minor
```

**Review by noranda — COMMENTED** [review]
Looks great! Just a few minor suggested changes and this is good to go! Thanks for the contribution! 👍

**Comment by jmargolisvt** [comment]
Thanks for all the suggestions! 😃

**Review by noranda — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]
thank you for contributing! 🎉

**Comment by jmargolisvt** [comment]
Thanks for the reviews!  I'm not authorized to merge this.  Would someone please do that for me? TIA!

**Pull request merged**